### ISSUE TITLE
EpubCFI.compare() returns wrong result

### ISSUE THREAD

**Issue by mrnerdy** [futurepress/epub.js#902]
The following code obviously should return `-1`, but it's returning `1`. Any idea why?

```
epubcfi.compare(
  'epubcfi(/6/16[id42]!/4[5N3C0-8c483216e03a4ff49927fc1a97dc7b2c]/10/1:317)',
  'epubcfi(/6/16[id42]!/4[5N3C0-8c483216e03a4ff49927fc1a97dc7b2c]/10/2[page18]/1:0)'
);
```

Another example:
```
epubcfi.compare(
  'epubcfi(/6/16[id42]!/4[5N3C0-8c483216e03a4ff49927fc1a97dc7b2c]/12/1:0)',
  'epubcfi(/6/16[id42]!/4[5N3C0-8c483216e03a4ff49927fc1a97dc7b2c]/12/2/1:9)'
);
```

**Comment by viking2917** [comment]
From the code: (the compare function returns 1 from the line quoted below). 

So, presumably it's because the first CFI in your example is less specific (shorter) than the second one, with less path elements. 

I'm no ePub expert so not sure if that is "correct" behavior or not, but that's what's going on. 


```
	// All steps in First equal to Second and First is Less Specific
			if (stepsA.length < stepsB.length) {
				return 1;
			}

```

**Comment by mrnerdy** [comment]
Yes, you are right, that's where it returns 1. And no, it's not correct.

**Comment by viking2917** [comment]
in the "for what it's worth" category, I tried the Readium EPUBcfi.compareCFIs function and it returns -1, as expected. 

(actually it returns an array of [-1, -1], their function compares start and end points of a cfi range).

**Comment by Macket** [comment]
I have the same. My example:
`const EpubCFI = new epub.CFI();`
`EpubCFI.compare('epubcfi(/6/2[html]!/4/10/2/1:0)', 'epubcfi(/6/2[html]!/4/10/1:196)')`
This code returns 1, but -1 expected

**Comment by fchasen** [comment]
Thanks for finding this, I'll fix the check asap

**Comment by fchasen** [comment]
Fixed in 0.3.84
